server/src/utils/validate.ts

```diff
@@ -0,0 +1,67 @@
+import { firstUC } from ".";
+import { ValidationBody } from "../types";
+
+function validateEmail(email) {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(String(email).toLowerCase());
+}
+
+export default (body: Array<ValidationBody>): { errors: object; values?: object } => {
+  const errors = {};
+  let values = {};
+
+  if (!body.length) return { errors };
+
+  if (body.length) {
+    values = body.map(({
+      config, key, value, type,
+    }) => {
+      const fieldName = config.title || firstUC(key);
+      value = value?.trim();
+      switch (type) {
+        case "email":
+          if (config.required) {
+            if (value.length === 0) {
+              errors[key] = `${fieldName} is required.`;
+            }
+          }
+
+          if (!errors[key]) {
+            const isValid = validateEmail(value);
+            if (!isValid) {
+              errors[key] = "Please provide valid email address";
+            }
+          }
+          break;
+        case "password":
+        case "text":
```

Comment:
case email can also be combined with this, unless you have something diff planned for email case